File: test_go_samples/edge_negatives.go
package main

// Negative tests — items that should NOT be captured as symbols.
// NoWreck captures only top-level declarations: functions, structs,
// interfaces, type aliases, and methods.
// Members (fields, methods on interfaces, constants) are NOT captured.

// Interface methods are NOT captured as symbols
type Repository interface {
    Find(id uint64) string
    Save(data string) bool
    Delete(id uint64)
}

// Struct fields are NOT captured as symbols
type Article struct {
    Title   string
    Content string
    Views   uint64
}

// Constants are NOT captured as symbols
const (
    MaxRetries = 3
    Timeout    = 30
)

// Variables are NOT captured as symbols
var globalConfig = Config{}

// Nested functions inside a function — NOT captured
func outer() {
    helper := func() int {
        return 42
    }
    _ = helper()
}

// Interface method implementations ARE captured (as methods),
// but the interface itself is separate.
func (a *Article) Summary() string {
    return a.Title
}

// Method on a non-struct type (type alias receiver) IS captured
type MyString string

func (s MyString) Upper() MyString {
    return MyString("UPPER")
}

File: test_rust_samples/edge_negatives.rs
// Negative tests — items that should NOT be captured as symbols.
// NoWreck captures only top-level declarations: functions, structs,
// traits, enums, type aliases, and impl methods.
// Members (fields, variants, trait method signatures) are NOT captured.

trait Repository {
    // These method signatures are NOT captured:
    fn find(&self, id: u64) -> Option<String>;
    fn save(&self, data: &str) -> bool;
    fn delete(&self, id: u64);
}

struct Article {
    // These fields are NOT captured:
    title: String,
    content: String,
    views: u64,
}

enum Color {
    // These variants are NOT captured:
    Red,
    Green,
    Blue,
    Named { r: u8, g: u8, b: u8 },
}

// Nested functions inside a function body — NOT captured
fn outer() {
    fn helper() -> i32 {
        42
    }
    let _ = helper();
}

// Items inside an impl block ARE captured (as methods),
// but the impl block itself is NOT a symbol.
impl Article {
    fn summary(&self) -> String {
        format!("{}: {}", self.title, &self.content[..50])
    }
}

// Items inside mod blocks are NOT captured (only top-level)
mod internal {
    pub fn secret() -> i32 { 42 }
    pub struct Hidden { pub value: i32 }
}
